pkg/commands/resolver.go

```diff
@@ -224,10 +224,15 @@ func makePublisher(po *options.PublishOptions) (publish.Interface, error) {
 		// If not publishing, at least generate a digest to simulate
 		// publishing.
 		if len(publishers) == 0 {
-			publishers = append(publishers, nopPublisher{
-				repoName: repoName,
-				namer:    namer,
-			})
+			noop, err := publish.NewNoOp(repoName,
```

Comment:
Unless the intention here is to export more functionality around no-op publishing, I'd kind of like to avoid increasing the public surface area of this change. We can do that just by passing more fields to `nopPublisher` and not exporting anything new.

If the intention _is_ to export more no-op publishing functionality, I'd like to understand more about the use case so we can make sure this change helps the issue.

In any case I think we can get this in before v0.12 and the repo move, if you're interested.

Reply:
I don't think there is any further no-op publishing functionality to export.

Reply:
What I'm saying is, this change exports new methods in `pkg/publish` to enable no-op publishing. But that's not necessary just to get the CLI behavior you're proposing -- we could do that just by passing more options into the (unexported) `nopPublisher` struct.

I think we should try to get this CLI behavior change done without exporting new methods, since once we export it, we're married to it forever. AFAIK there aren't a lot of use cases for no-op publishing, vs local publishing or tarball publishing, which are better supported don't-push-to-a-registry options, except in this very narrow use case.